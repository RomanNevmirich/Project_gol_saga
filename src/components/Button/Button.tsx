import React from 'react';
import './button.module.css';

export interface ButtonProps {
  /**
   * Is this the principal call to action on the page?
   */
  primary?: boolean;
  /**
   * What background color to use
   */
  backgroundColor?: string;
  /**
   * How large should the button be?
   */
  size?: 'small' | 'medium' | 'large';
  /**
   * Button contents
   */
  label: string;
  id: string;
  /**
   * Optional click handler
   */
  onClick?: () => void;
}

/**
 * Primary UI component for user interaction
 */
export const Button = ({
  primary = false,
  size = 'medium',
  backgroundColor,
  label,
  id,
  ...props
}: ButtonProps) => {
  const mode = primary ? 'storybook-button--primary' : 'storybook-button--secondary';
  return (
    <button
      data-testid={id}
      type="button"
      className={['storybook-button', `storybook-button--${size}`, mode].join(' ')}
      style={{ backgroundColor }}
      {...props}
    >
      {label}
    </button>
  );
};
