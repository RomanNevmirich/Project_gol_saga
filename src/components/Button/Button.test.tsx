import React from 'react';
import { render, screen, fireEvent, createEvent } from '@testing-library/react';
import { Button } from './Button';

describe('Button component check', () => {
    it('Button render', () => {
        const { asFragment } = render(
          <Button
            id='test'
            onClick={() => null}
            label='button'
            primary
            size='small'
          />
        );
        screen.debug();
        expect(asFragment()).toMatchSnapshot();
      });

      it('Button click', () => {
        const onclick = jest.fn();
        render(
          <Button id='test' onClick={onclick} label='label' primary />
        );
        const button = screen.getByTestId('test');
        fireEvent.click(button);
        expect(onclick).toBeCalled();
      });

      it('Button label', () => {
        render(
          <Button
            id='test'
            onClick={() => null}
            label='button1'
            primary
            size='small'
          />
        );
        const button = screen.getByTestId('test');
        console.log(button.textContent);
        expect(button.textContent === 'button1').toBe(true);
      });
});