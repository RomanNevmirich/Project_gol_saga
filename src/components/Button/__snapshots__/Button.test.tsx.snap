// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component check Button render 1`] = `
<DocumentFragment>
  <button
    class="storybook-button storybook-button--small storybook-button--primary"
    data-testid="test"
    type="button"
  >
    button
  </button>
</DocumentFragment>
`;
